src/solvers/lowering/byte_operators.cpp

```diff
@@ -233,16 +233,16 @@ exprt lower_byte_extract(const byte_extract_exprt &src, const namespacet &ns)
     const typet &subtype=array_type.subtype();
 
     auto element_width = pointer_offset_bits(subtype, ns);
-    mp_integer num_elements;
+    const auto num_elements = numeric_cast<mp_integer>(array_type.size());
```

Comment:
I can't spot the bug you've fixed here that causes the test to fail - could you pull it out into a standa lone commit for ease of reviewing and when that test envitably gets `git blame`d to see why it is turned off.

Reply:
Done, it's now the first (very small) commit.